modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -527,6 +531,463 @@ public void testAddCustomObjectField() throws Exception {
+				"isEmpty(able)"
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _addCustomObjectField(
+			new AggregationObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					new ObjectFieldSettingBuilder(
+					).name(
+						"function"
+					).value(
+						"MAX"
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"objectFieldName"
+					).value(
+						"integerObjectFieldName"
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"objectRelationshipName"
+					).value(
+						objectRelationship.getName()
+					).build())
+			).readOnly(
+				ObjectFieldConstants.READ_ONLY_FALSE
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _addCustomObjectField(
+			new AggregationObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					new ObjectFieldSettingBuilder(
+					).name(
+						"function"
+					).value(
+						"MAX"
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"objectFieldName"
+					).value(
+						"integerObjectFieldName"
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"objectRelationshipName"
+					).value(
+						objectRelationship.getName()
+					).build())
+			).readOnly(
+				invalidReadOnly
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		// readOnly with Proxy
+
+		objectDefinition1 =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false, false,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Test", null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				true, ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE,
+				Collections.emptyList());
+
+		objectField = _objectFieldLocalService.getObjectField(
+			objectDefinition1.getObjectDefinitionId(), "creator");
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
```

Comment:
This is a repeated code snippet, see lines 595 to 601.
I would suggest you change creating a method or avoid the assert calls since you already know the results.